shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -340,18 +341,17 @@ View onCreateView(
                 + " RenderMode.surface.");
       }
 
-      flutterView
-          .getViewTreeObserver()
-          .addOnPreDrawListener(
-              new OnPreDrawListener() {
-                @Override
-                public boolean onPreDraw() {
-                  if (isFlutterUiDisplayed) {
-                    flutterView.getViewTreeObserver().removeOnPreDrawListener(this);
-                  }
-                  return isFlutterUiDisplayed;
-                }
-              });
+      activePreDrawListener =
+          new OnPreDrawListener() {
+            @Override
+            public boolean onPreDraw() {
+              if (isFlutterUiDisplayed) {
```

Comment:
I suspect this might run again after you rotate the screen. We might want to run this only once?

Reply:
I tested it locally and registering the callback only happens once. We also deregister the listener the moment `isFlutterUiDisplayed` returns true, so it should be safe.